hadoop-ozone/tools/src/test/java/org/apache/hadoop/ozone/debug/datanode/container/TestChecksumSubcommand.java

```diff
@@ -0,0 +1,214 @@
+
+    File treeFile = new File(tempDir.toFile(), CONTAINER_ID + ".tree");
+
+    JsonNode actualJson = runChecksumCommand(treeFile);
+
+    // Verify the structure and key fields
+    assertThat(actualJson.isArray()).isTrue();
+    assertThat(actualJson.size()).isEqualTo(1);
+
+    JsonNode containerJson = actualJson.get(0);
+    assertThat(containerJson.get("containerID").asLong()).isEqualTo(CONTAINER_ID);
+    assertThat(containerJson.get("filePath").asText()).isEqualTo(treeFile.getAbsolutePath());
+    assertThat(containerJson.has("containerMerkleTree")).isTrue();
+
+    JsonNode merkleTree = containerJson.get("containerMerkleTree");
+    assertThat(merkleTree.has("dataChecksum")).isTrue();
+    assertThat(merkleTree.has("blockMerkleTrees")).isTrue();
+    assertThat(merkleTree.get("blockMerkleTrees").isArray()).isTrue();
+    assertThat(merkleTree.get("blockMerkleTrees").size()).isEqualTo(5); // Default buildTestTree creates 5 blocks
+  }
+
+  @Test
+  void testChecksumCommandWithNonExistentFile() throws Exception {
+    ChecksumSubcommand command = new ChecksumSubcommand();
+    command.setTreeFilePath("/non/existent/file.tree");
+
+    RuntimeException exception = assertThrows(RuntimeException.class, command::call);
+
+    assertThat(err.toString(DEFAULT_ENCODING)).contains("Error: Tree file does not exist");
+    assertThat(exception.getMessage()).contains("Tree file does not exist");
+  }
+
+  @Test
+  void testChecksumCommandWithCorruptedFile() throws Exception {
+    // Create a corrupted tree file
+    File treeFile = new File(tempDir.toFile(), "corrupted.tree");
+    try (OutputStream fos = Files.newOutputStream(treeFile.toPath())) {
+      fos.write(new byte[]{1, 2, 3, 4, 5}); // Invalid protobuf data
+    }
+
+    ChecksumSubcommand command = new ChecksumSubcommand();
+    command.setTreeFilePath(treeFile.getAbsolutePath());
+
+    RuntimeException exception = assertThrows(RuntimeException.class, command::call);
+
+    assertThat(err.toString(DEFAULT_ENCODING)).contains("Error reading tree file");
+    assertThat(exception.getMessage()).contains("Failed to read tree file");
+  }
+
+  @Test
+  void testChecksumCommandWithEmptyFile() throws Exception {
+    // Create an empty tree file
+    File treeFile = new File(tempDir.toFile(), "empty.tree");
+    ContainerProtos.ContainerChecksumInfo emptyInfo = ContainerProtos.ContainerChecksumInfo.newBuilder().build();
+    try (OutputStream fos = Files.newOutputStream(treeFile.toPath())) {
+      emptyInfo.writeTo(fos);
+    }
+
+    JsonNode actualJson = runChecksumCommand(treeFile);
+    
+    // Build expected JSON string for empty container
+    String expectedJson = String.format(
+        "[ {" +
+        "\"containerID\" : 0," +
+        "\"filePath\" : \"%s\"" +
+        "} ]", 
+        treeFile.getAbsolutePath().replace("\\", "\\\\")
+    );
+    ObjectMapper mapper = JsonUtils.getDefaultMapper();
+    JsonNode expectedJsonNode = mapper.readTree(expectedJson);
```

Comment:
We should probably just build the expected `JsonNode` instead of creating this string and then converting it.